art/attacks/evasion/laser_attack/laser_attack.py

```diff
@@ -0,0 +1,263 @@
+# %%
+from typing import Optional, Tuple, Union, List
+import logging
+import numpy as np
+from art.attacks.evasion.laser_attack.utils import \
+    AdversarialObject, AdvObjectGenerator, ImageGenerator, \
+    DebugInfo, Line, wavelength_to_RGB
+from art.attacks.attack import EvasionAttack
+from art.attacks.evasion.laser_attack.algorithms import greedy_search
+
+logger = logging.getLogger(__name__)
+
+# %%
+class LaserAttack(EvasionAttack):
+    attack_params = EvasionAttack.attack_params + [
+        "optimizer",
+        "iterations",
+        "random_initializations",
+        "tensor_board, bool] = False",
+        "actual_class",
+        "actual_class_confidence"
+    ]
+    AdvParams = Tuple[Optional[AdversarialObject], Optional[int]]
+    _estimator_requirements = ()
+
+    def __init__(
+        self,
+        estimator,
+        iterations: int,
+        laser_generator: AdvObjectGenerator,
+        image_generator: ImageGenerator = ImageGenerator(),
+        random_initializations: int = 1,
+        optimisation_algorithm = greedy_search,
+        tensor_board: Union[str, bool] = False,
+        debug: Optional[DebugInfo] = None
+    ) -> None:
+
+        super().__init__(estimator=estimator, tensor_board=tensor_board)
+        self.iterations = iterations
+        self.random_initializations = random_initializations
+        self.optimisation_algorithm = optimisation_algorithm
+        self.__laser_generator = laser_generator
+        self.__image_generator = image_generator
+        self.__debug = debug
+
+        self._check_params()
+
+    def generate(
+        self,
+        x: np.ndarray,
+        *args,
+        **kwargs
+    ) -> Optional[List]:
```

Comment:
Method `generate` should return a `np.ndarray` of the adversarial examples, would this be possible?

Reply:
Yes, sure. I wanted to return adversarial parameters but I'll do it in an other function.